src/backends/sqlite3/CMakeLists.txt

```diff
@@ -1,7 +1,9 @@
 include(soci_define_backend_target)
 
 if (SOCI_SQLITE3_AUTO)
```

Comment:
@Krzmbrzl Somehow this test doesn't seem to work when rerunning CMake. I.e. it works fine when running it the first time or with `--fresh` later but not when rerunning it without that option, including when `cmake --build` reruns it automatically. I've tried to understand what's going on here, but the mechanics of defining `${BACKEND}_AUTO` are too confusing, so I'd be grateful if you could please have a look at this. TIA!

Reply:
OK, so this seems to happen because we intentionally override the cached variable with `ON`. I don't think we need to do this, so I'll create a separate PR to stop doing it.